testsuite/model/src/test/java/org/keycloak/testsuite/model/storage/tree/sample/Dict.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.testsuite.model.storage.tree.sample;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.keycloak.models.map.client.MapClientEntity;
+import org.keycloak.models.map.client.MapClientEntityFieldDelegate;
+import org.keycloak.models.map.client.MapClientEntityFields;
+import org.keycloak.models.map.common.EntityField;
+import org.keycloak.models.map.common.UpdatableEntity;
+import org.keycloak.models.map.common.delegate.EntityFieldDelegate;
+import org.keycloak.models.map.common.delegate.HasEntityFieldDelegate;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public class Dict<E> extends UpdatableEntity.Impl implements EntityFieldDelegate<E> {
```

Comment:
Is there some reason why this test is part of model tests? Can this be part of unit tests in the `map` module?

Reply:
This class serves as a cheap example of a LDAP or Yaml store. It will be used in the tree store which will be tested in model tests. So it is intentionally in the model tests.